### ISSUE TITLE
3.0 - Incorrect tax amount being charged

### ISSUE THREAD

**Issue by ashleyfae** [awesomemotive/easy-digital-downloads#9054]
### Expected behavior
When I check out using a country code that does not have a tax rate, I expect to not be charged tax.

### Actual behavior
I am charged tax.

### Steps to reproduce the behavior
I'm still solidifying steps, but I did manage to reproduce it once at least:

1. Set your shop's base country. Mine is set to GB.
2. Set up a tax rate for your shop's base country. Do not have any other tax rates.
3. Check out via a fresh account and select a _different_ country. So I'm checking out as US.
4. Checkout shows $0 tax. For me it's showing $35 as the total (which is the price without tax).
4. Not sure if it's important, but check out using the Stripe gateway.
5. Purchase receipt shows that the GB tax rate was charged for me ($7 tax, total price $42.00).
6. Stripe charged $42.

tl;dr: In this case I should NOT have been charged tax, but I was.

### Information (if a specific version is affected):
PHP Version:

EDD Version (or branch): release/3.0

WordPress Version:

Any other relevant information:

**Comment by ashleyfae** [comment]
So far:

- Reproduced using Stripe gateway.
- Did same steps using the store gateway and did not get the problem.
- `edd_get_cart_tax_rate()` in Stripe is returning `0`
- `edd_get_cart_tax()` is returning an amount

**Comment by ashleyfae** [comment]
Opting for a fix in the Stripe gateway https://github.com/awesomemotive/edd-stripe/issues/736

This problem is because the Stripe gateway is doing some funky stuff and then spoofing `$_POST`. The tax rate cache gets saved in between the ajax action loading and the `$_POST` getting filled, which I'm hoping isn't a common thing in other gateways.
